app/models/energy/constructionset.py

```diff
@@ -0,0 +1,146 @@
+"""Construction Set Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.construction import EnergyConstructionOpaque, EnergyConstructionWindow
+
+
+class WallSet(BaseModel):
+    """A set of constructions for wall assemblies."""
+
+    type: Enum('WallSet', {
+               'type': 'WallSet'})
+
+
+    interior_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    ground_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+
+class FloorSet(BaseModel):
+    """A set of constructions for floor assemblies."""
+
+    type: Enum('FloorSet', {
+               'type': 'FloorSet'})
+
+
+    interior_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    ground_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+class RoofCeilingSet(BaseModel): 
+    """A set of constructions for roof and ceiling assemblies."""
+
+    type: Enum('RoofCeilingSet', {
+               'type': 'RoofCeilingSet'})
+
+    interior_ceiling: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_roof: EnergyConstructionOpaque = Schema(
+        default=None
+    )    
+
+    ground_ceiling: EnergyConstructionOpaque = Schema(
```

Comment:
I know my comment above means that this one will be changed to `ground_construction` but I just wanted to highlight that `ground_ceiling` is a particular misnomer since `ceiling` implies that the construction is on the interior of a building, which is not the case here.  This construction is meant to be applied to the roof of a basement that is covered by the ground (like a bunker).  For the case of an interior ceiling that is for a below-grade floor, the `interior_construction` should be used.  I know this doesn't mean too much at this point but this will be important to keep in mind for the energy-model-measure.